src/components/refactor/dialogs/generator/set-points/voltage-regulation.js

```diff
@@ -23,20 +23,52 @@ import {
 import RegulatingTerminalForm from '../../regulating-terminal/regulating-terminal-form';
 import { Box } from '@mui/system';
 import Grid from '@mui/material/Grid';
-import { FormattedMessage } from 'react-intl';
-import { useWatch } from 'react-hook-form';
+import { FormattedMessage, useIntl } from 'react-intl';
+import { useFormContext, useWatch } from 'react-hook-form';
 
 const VoltageRegulation = ({
     studyUuid,
     currentNodeUuid,
     voltageLevelOptions,
+    generatorInfos,
 }) => {
+    const intl = useIntl();
+    const { setValue } = useFormContext();
+    function getPreviousRegulationType(generatorInfos) {
+        if (generatorInfos?.voltageRegulatorOn) {
+            return generatorInfos?.regulatingTerminalVlId ||
+                generatorInfos?.regulatingTerminalConnectableId
+                ? REGULATION_TYPES.DISTANT
+                : REGULATION_TYPES.LOCAL;
+        } else {
+            return null;
+        }
+    }
+    const previousRegulationTypeLabel = getPreviousRegulationType(
+        generatorInfos
+    )?.label
+        ? intl.formatMessage({
+              id: getPreviousRegulationType(generatorInfos)?.label,
+          })
+        : undefined;
+
     const voltageRegulationType = useWatch({
         name: VOLTAGE_REGULATION_TYPE,
     });
 
+    useEffect(() => {
+        if (
+            generatorInfos?.regulatingTerminalVlId ||
+            generatorInfos?.regulatingTerminalConnectableId
+        ) {
+            setValue(VOLTAGE_REGULATION_TYPE, REGULATION_TYPES.DISTANT.id);
+        }
+    }, [generatorInfos, setValue]);
```

Comment:
if think it can be handled in the form default values

Reply:
yes i changed